user_modules/shared_testcases/commonFunctions.lua

```diff
@@ -1230,4 +1231,59 @@ function commonFunctions:update_json_file(path_to_json, old_section, new_section
   file:close()
 end
 
+-- ---------------------------------------------------------------------------------------------
+--27. Function joins paths of file system
+-- ---------------------------------------------------------------------------------------------
+--! @brief Return the path resulting from combining the individual paths or nil
+--! if the second (or later) path is absolute will be returned the last absolute path (joined with any non-absolute paths following).
+--! empty elements (except the first) will be ignored.
+--! @string p1 A file path
+--! @string p2 A file path
+--! @string ... more file paths
+function commonFunctions:path_join(p1, p2, ...)
+
+    local function assertPath(val)
+        if type(val) ~= "string" or string.find(val, "%c") then
+            return false
+        end
+        return true
+    end
+
+    if assertPath(p1) and assertPath(p2) then
+        if select('#',...) > 0 then
+            local p = commonFunctions:path_join(p1,p2)
+            local args = {...}
+            for i = 1, #args do
+                assertPath(args[i])
+                p = commonFunctions:path_join(p,args[i])
+            end
+            return p
+        end
+
+        -- second path is absolute
+        if string.sub(p2,1,1) == '/' then
+            return p2
+        end
+
+        if p1 == "" then
+            return nil
+        end
+
+        local endChar = string.sub(p1,#p1,#p1)
+        if endChar ~= "/" then
+            p1 = p1 .. "/"
+        end
+
+        if string.sub(p2, 1, 2) == "./" then
+            p2 = string.sub(p2, 3)
+        end
+
+        p2 = string.gsub(p2, "%s*$", "")
+
+        return string.gsub(p1 .. p2, "//", "/")
+    else
+        return nil
+    end
+end
+
```

Comment:
@aderiabin, I propose to have simpler solution:
1. Check, that all parameters are valid (there are no specific characters, that are not allowed for paths)
2. In loop (by params)
a) Remove '.' from param_i
b) add '/' at beginning and at the end of param_i
c) concatenate param_i with result string
3) Remove '//' from the result string in all matches